### PR TITLE
KAFKA-7880:Naming worker thread by task id

### PR DESCRIPTION
**Pull request by FreeeeLy**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by FreeeeLy** [comment]
@ewencp @rhauch @kkonstantine - Can You or someone from your team please review this PR?

Thanks

**Comment by FreeeeLy** [comment]
@mjsax, @guozhangwang - Can You or someone from your team please review this PR?

Thanks.

**Review by mageshn — COMMENTED** [review]
@FreeeeLy Thanks a lot for the PR. Overall, I believe this will be pretty useful and the changes LGTM

**Review by mageshn — APPROVED** [review]

**Comment by thatb96** [comment]
Thanks

**Comment by FreeeeLy** [comment]
retest this please

**Comment by FreeeeLy** [comment]
Java11 had one unrelated test failure:
* `kafka.api.SaslSslAdminClientIntegrationTest.testMinimumRequestTimeouts`

Java8 had one unrelated test failure:

* `kafka.api.AdminClientIntegrationTest.testMinimumRequestTimeouts`

Retest this, please.

**Comment by guozhangwang** [comment]
cc @rhauch

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java
Comment (kkonstantine) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java]: I wonder if `task-thread-` is preferable since it's a smaller departure from the previous prefix that was `pool-<poolId>-thread-<threadId>-`. I'm a bit on the fence, but also not too important.

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java
Comment (kkonstantine) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java]: `toString` is not required.

**Review by kkonstantine — APPROVED** [review]
Thanks @FreeeeLy
The changes make sense to me. Even though restoring the thread id is not strictly required, it's a better pattern to follow, so I'm in favor. I have two minor comments, but otherwise LGTM

**Comment by kkonstantine** [comment]
And one more thing I forgot. A brief explanation of the change on the PR that will go in the commit message would be helpful. Thanks again!

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java
Comment (FreeeeLy) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java]: Thanks, I agree `task-thread` is more preferable

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java
Comment (FreeeeLy) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java]: done

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java
Comment (FreeeeLy) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java]: done

**Comment by FreeeeLy** [comment]
Java11 passed

Java8 had one unrelated test failure:
`kafka.api.ConsumerBounceTest.testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup`

I have retested this several times but they may have some unrelated test failures.
How can I solve this kind of problems?

**Review by kkonstantine — APPROVED** [review]
Thank you for addressing my comments @FreeeeLy !
LGTM by me. (I wouldn't block this PR if indeed the test failures are unrelated).

**Comment by FreeeeLy** [comment]
cc @rhauch

**Comment by FreeeeLy** [comment]
Retest this, please.

**Comment by FreeeeLy** [comment]
Retest this, please

**Review by rhauch — APPROVED** [review]
LGTM. Thanks, @FreeeeLy!

**Comment by rhauch** [comment]
My apologies for not including @kkonstantine and @mageshn on the "Reviewers" comment in the squashed commit message. Thank you both, and I'll not make the same mistake again.

**Pull request merged**
